pkg/knowledge_base2/operational_rule.go

```diff
@@ -16,6 +16,23 @@ type (
 		ConfigurationRules []ConfigurationRule `json:"configuration_rules" yaml:"configuration_rules"`
 	}
 
+	EdgeRule struct {
+		If                 string                `json:"if" yaml:"if"`
+		Steps              []EdgeOperationalStep `json:"steps" yaml:"steps"`
+		ConfigurationRules []ConfigurationRule   `json:"configuration_rules" yaml:"configuration_rules"`
+	}
+
+	PropertyRule struct {
+		If    string           `json:"if" yaml:"if"`
+		Step  *OperationalStep `json:"step" yaml:"step"`
```

Comment:
yeah removing the resource is coming in the next refactor where we introduce edgeStep thats just resource + operational step.

i was doing a pointer to tell if it exists or not, rather than checking against OperationalStep{} each time

Reply:
I wouldn't check against `OperationalStep{}` because that allocates. I'd just do the length check in my original comment. Generally I try to avoid pointers unless there's a good reason to, like if copying the value should share the field. I think we _don't_ want that behaviour so that'd be a good reason not to use a pointer.

This is a nit though, non-blocking